### PR TITLE
Add htmx and hx-boost for header navs

### PR DESCRIPTION
**Pull request by theskumar**
- Add htmx integration
- Add hx-boost for header nav; it degrades gracefully! 

Configure https://github.com/rstacruz/nprogress to show progress on ajax activities.

Note: django-htmx is not used right now but will be used to add more htmx-based functionalities.

### CODE REVIEW

**Comment by theskumar** [comment]
@frjo The CSS for nprogress is provided by the library. I see it doesn't lint with our style linter. What would you like to do with these? Ideally, I would like to skip all these issues. 

```
hypha/static_src/src/sass/apply/components/_nprogress.scss
  1:1   ✖  Expected "#nprogress" to have no more than 0 ID selectors                                    selector-max-id
  5:5   ✖  Expected "#nprogress .bar" to have no more than 0 ID selectors                               selector-max-id
  6:21  ✖  Unexpected hex color "#0076ff"                                                               color-no-hex
 18:5   ✖  Expected "#nprogress .peg" to have no more than 0 ID selectors                               selector-max-id
 21:17  ✖  Unexpected unit                                                                              length-zero-no-unit
 24:30  ✖  Unexpected hex color "#0076ff"                                                               color-no-hex
 24:47  ✖  Unexpected hex color "#0076ff"                                                               color-no-hex
 27:44  ✖  Unexpected unit                                                                              length-zero-no-unit
 31:5   ✖  Expected "#nprogress .spinner" to have no more than 0 ID selectors                           selector-max-id
 39:5   ✖  Expected "#nprogress .spinner-icon" to have no more than 0 ID selectors                      selector-max-id
 45:27  ✖  Unexpected hex color "#29d"                                                                  color-no-hex
 46:28  ✖  Unexpected hex color "#29d"                                                                  color-no-hex
 57:5   ✖  Expected ".nprogress-custom-parent #nprogress .spinner" to have no more than 0 ID selectors  selector-max-id
 57:5   ✖  Expected ".nprogress-custom-parent #nprogress .bar" to have no more than 0 ID selectors      selector-max-id
```

**Comment by theskumar** [comment]
@frjo I've disabled the stylelint to be run on the `/components/_nprogress.scss` https://github.com/HyphaApp/hypha/pull/3226/commits/b56f152454ff0fb3b9ec84a7c447e3b1c3a03150 hope that's okay.

File: package.json [hunk 1]
```diff
@@ -5,6 +5,9 @@
     "author": "Fredrik Jonsson",
     "license": "GPL-2.0",
     "babel": {
+        "exclude": [
+            "**/*.min.js"
+        ],
```
Comment (theskumar) [inline package.json:10]: Required for nprogress.js to work locally.

Comment (frjo) [reply package.json:10]: And not really any point in processing minified libs.

Comment (theskumar) [reply package.json:10]: Right!

**Comment by frjo** [comment]
@theskumar Thanks for pushing this! I'm especially looking forward to applying this to the submission detail page and project detail page in the future.

Is the pages with hx-boost supposed to auto reload every 30s?

What are the main performance benefits?

If I understand correctly body gets reloaded and not the head with all js+css etc.? Are these resources not mostly cached by the browser in any case?

**Comment by theskumar** [comment]
`hx-boost` doesn't auto-reload. It makes an ajax call to fetch the page's content and replaces the current DOM with it. 

`hx-boost` prevents a complete teardown of the DOM and building from scratch. This makes the new page render considerably fast and makes it feel like SPAs. This gives us a quick win without introducing any complications.

Yes. It does some caching and would display a preview of the page before the page is loaded.

**Comment by frjo** [comment]
The reloading is from PR #3224, forgot about that one.

**Pull request merged**